pymc/distributions/multivariate.py

```diff
@@ -1631,11 +1592,95 @@ def logp(value, n, eta):
         )
 
 
-@_default_transform.register(LKJCorr)
+@_default_transform.register(_LKJCorr)
 def lkjcorr_default_transform(op, rv):
     return MultivariateIntervalTransform(floatX(-1.0), floatX(1.0))
 
 
+# Thin wrapper around _LKJCorr
+class LKJCorr:
+    r"""
+    The LKJ (Lewandowski, Kurowicka and Joe) log-likelihood.
+
+    The LKJ distribution is a prior distribution for correlation matrices.
+    If eta = 1 this corresponds to the uniform distribution over correlation
+    matrices. For eta -> oo the LKJ prior approaches the identity matrix.
+
+    ========  ==============================================
+    Support   Upper triangular matrix with values in [-1, 1]
+    ========  ==============================================
+
+    Parameters
+    ----------
+    n : tensor_like of int
+        Dimension of the covariance matrix (n > 1).
+    eta : tensor_like of float
+        The shape parameter (eta > 0) of the LKJ distribution. eta = 1
+        implies a uniform distribution of the correlation matrices;
+        larger values put more weight on matrices with few correlations.
+    return_matrix : bool, default=False
+        If True, returns the full correllation matrix.
+        False only returns the values of the upper triangular matrix excluding
+        diagonal in a single vector of length n(n-1)/2 for backwards compatibility
+
+    Notes
+    -----
+    This is mainly useful if you want the standard deviations to be fixed, as
+    LKJCholsekyCov is optimized for the case where they come from a distribution.
+
+    Examples
+    --------
+    .. code:: python
+
+        with pm.Model() as model:
+
+            # Define the vector of fixed standard deviations
+            sds = 3*np.ones(10)
+
+            corr = pm.LKJCorr(
+                'corr', eta=4, n=10, return_matrix=True
+            )
+
+            # Define a new MvNormal with the given correllation matrix
+            vals = sds*pm.MvNormal('vals', mu=np.zeros(10), cov=corr, shape=10)
+
+            # Or transform an uncorrelated normal distribution:
+            vals_raw = pm.Normal('vals_raw', shape=10)
+            chol = pt.linalg.cholesky(corr)
+            vals = sds*pt.dot(chol,vals_raw)
+
+
+    References
+    ----------
+    .. [LKJ2009] Lewandowski, D., Kurowicka, D. and Joe, H. (2009).
+        "Generating random correlation matrices based on vines and
+        extended onion method." Journal of multivariate analysis,
+        100(9), pp.1989-2001.
+    """
+
+    def __new__(cls, name, n, eta, *, return_matrix=False, **kwargs):
+        if not return_matrix:
+            return _LKJCorr(name, eta=eta, n=n, **kwargs)
+        else:
+            c_vec = _LKJCorr(name + "_raw", eta=eta, n=n, **kwargs)
+            return pm.Deterministic(name, cls.vec_to_corr_mat(c_vec, n))
```

Comment:
Not sure about the name change and wrapping in a Deterministic by default, but not against it either

Reply:
The argument for having the packed form is that it saves memory. This actually more than doubles it because we save both the packed form and the dense one. We could leave to the users to wrap in a Deterministic if they need to have it after sampling.

I would add a docstring code example showing how to do it.

Reply:
Maybe check the discussion #6828 about setting internal deterministics/coords?

Reminds me I need to finish that PR